obspy/imaging/ray_paths.py

```diff
@@ -0,0 +1,594 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Plotting 3D ray paths.
+
+:author:
+    Matthias Meschede
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+from colorsys import hls_to_rgb, rgb_to_hls
+
+import warnings
+import numpy as np
+from matplotlib.colors import hex2color
+
+
+def plot_rays(inventory=None, catalog=None, station_latitude=None,
+              station_longitude=None, event_latitude=None,
+              event_longitude=None, event_depth_in_km=None, phase_list=('P',),
+              kind='mayavi', colorscheme='default', animate=False,
+              savemovie=False, figsize=(800, 800), coastlines='internal',
+              taup_model='iasp91', icol=0, event_labels=True,
+              station_labels=True, fname_out=None, view_dict=None):
+    """
+    Ray path plotting routine
+
+    :param inventory: an obspy station inventory
+    :param catalog: an obspy event catalog
+    :param station_latitude: station latitude (alternative to inventory)
+    :param station_longitude: station longitude (alternative to inventory)
+    :param event_latitude: event latitude (alternative to catalog)
+    :param event_longitude: event longitude (alternative to catalog)
+    :param event_depth_in_km: event depth in km (alternative to a catalog)
```

Comment:
I hesitate a bit using Origin etc. In my opinion the user should just pass catalogs and inventories. Otherwise it doesn't simplify the function. Catalog and inventory can contain a single station. A simple function is needed to generate such catalogs and inventories from scratch. If no network is given, it can be filled with a __ network that contains stations without networks. Same for the catalog.

```
inv  = inventory()
stat = station(lat=20, lon=30)
inv.append(stat)
```